contracts/VaultFactory.vy

```diff
@@ -172,15 +170,44 @@ def protocol_fee_config(vault: address = msg.sender) -> PFConfig:
     @return The protocol fee config for the msg sender.
     """
     # If there is a custom protocol fee set we return it.
-    if self.use_custom_protocol_fee[vault]:
+    config_data: uint256 = self.custom_protocol_fee_data[vault]
+    if self._unpack_custom_flag(config_data):
         # Always use the default fee recipient even with custom fees.
-        return PFConfig({
-            fee_bps: self.custom_protocol_fee[vault],
-            fee_recipient: self.default_protocol_fee_config.fee_recipient
-        })
+        return (
+            self._unpack_protocol_fee(config_data),
+            self._unpack_fee_recipient(self.default_protocol_fee_data)
+        )
     else:
         # Otherwise return the default config.
-        return self.default_protocol_fee_config
+        config_data = self.default_protocol_fee_data
+        return (
+            self._unpack_protocol_fee(config_data), 
+            self._unpack_fee_recipient(config_data)
+        )
+
+@view
+@external
+def use_custom_protocol_fee(vault: address) -> bool:
+    return self._unpack_custom_flag(self.custom_protocol_fee_data[vault])
+
+@view
+@internal
+def _unpack_protocol_fee(config_data: uint256) -> uint16:
+    return convert(shift(config_data, -8) & FEE_BPS_MASK, uint16)
+    
+@view
+@internal
+def _unpack_fee_recipient(config_data: uint256) -> address:
+    return convert(shift(config_data, -24), address)
+
+@view
+@internal
+def _unpack_custom_flag(config_data: uint256) -> bool:
+    return config_data & 1 == 1
+
+@internal
+def _pack_data(recipient: address, fee: uint16, custom: bool) -> uint256:
```

Comment:
small: 
could use a more descriptive function name like `_pack_fee_config_data` for clarity if/when more slot packing functions are needed in future.

Reply:
https://github.com/yearn/yearn-vaults-v3/pull/211/commits/abf3e20af1b37397efaa8569934ee92889cea712